### PR TITLE
Person.rb

### PR DESCRIPTION
**Pull request by markotuya0**
in this branch i have worked on the following things : 
 I add the person class and data,
I added the student and teacher class  and its data expended from the person class

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    @parent_permission = parent_permission
+  end
+
+  private_class_method def self.of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    @age >= 18 && @parent_permission
+  end
```
Comment (Bondok6) [inline person.rb]: - [ ] Great job here 💯 However, according to the requirements `can_use_services?` should return true if person is of age **or** if they have permission from parents. (not and), so Kindly fix this small issue 👍

image

File: README.md [hunk 2]
```diff
@@ -1 +1,61 @@
...
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+- Inspiration
+
+## 📝 License
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (Bondok6) [inline README.md:61]: - [ ] Your README file looks professional 💯 But I noticed that The MIT link is broken, so I suggest adding the `MIT.md` file to your project 👌, Here is the [Micorverse Template](https://github.com/microverseinc/readme-template/blob/master/MIT.md), and remember to customize it 😉

> The MIT license gives users express permission to reuse code for any purpose, sometimes even if the code is part of proprietary software. As long as users include the original copy of the MIT license in their distribution, they can make any changes or modifications to the code to suit their own needs. 💯

**Review by Bondok6 — CHANGES_REQUESTED** [review]
## Hi @markotuya0 🙋‍♂️, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No Linter errors ✅
- Correct GitHub flow ✅
- Your repo is professional ✅


## Required Changes ♻️
- [ ] _Check the comments under the review._ ⬇


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  private_class_method def self.of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    if of_age? || @parent_permission
+      true
+    else
+      false
+    end
```
Comment (FrederickMih) [inline person.rb:21]: - [**Optional**] Great job for using the correct logical operator issue raised by the previous reviewer :+1: I think there's no reason for explicitly returning true or false in this method. The method name itself(**can_use_services?**) return true by default else false if the statement is not true. what you have done in of_age? method can equally true in this method.

**Review by FrederickMih — APPROVED** [review]
# Approved :tada: 

Hi @markotuya0, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- Great job using the correct logic proposed by the reviewer. :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**